x/checkers/keeper/msg_server_reject_game.go

```diff
@@ -10,8 +12,35 @@ import (
 func (k msgServer) RejectGame(goCtx context.Context, msg *types.MsgRejectGame) (*types.MsgRejectGameResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Handling the message
-	_ = ctx
+	storedGame, found := k.Keeper.GetStoredGame(ctx, msg.IdValue)
+	if !found {
+		return nil, errors.New("Game not found " + msg.IdValue)
+	}
+	fullGame := storedGame.ToFullGame()
+
+	// Is it an expected player? And did the player already play?
+	if strings.Compare(storedGame.Red, msg.Creator) == 0 {
+		if 1 < fullGame.MoveCount {
+			return nil, errors.New("Red player has already played, and cannot reject")
+		}
+	} else if strings.Compare(storedGame.Black, msg.Creator) == 0 {
+		if 0 < fullGame.MoveCount {
+			return nil, errors.New("Black player has already played, and cannot reject")
+		}
+	} else {
+		return nil, errors.New("Message creator is not a player")
+	}
+
+	// Remove the game completely as it is not interesting to keep it.
+	k.Keeper.RemoveStoredGame(ctx, msg.IdValue)
+
+	// What to emit
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(types.RejectGameEventKey,
```

Comment:
see https://github.com/cosmos/b9-checkers-academy-draft/pull/5 for event construction